### PR TITLE
SMZ3: Add Start Inventory From Pool

### PR DESCRIPTION
**Pull request by dontjoome**
Original PR that I accidentally closed by deleting the branch associated with it - https://github.com/ArchipelagoMW/Archipelago/pull/3689

Reopened in the hopes of getting it added, since it's a nice thing to have so you don't find duplicates of stuff out in the wild.

### CODE REVIEW

**Comment by ScipioWright** [comment]
You should add the game name to the start of the title

**Comment by ScipioWright** [comment]
Also you are changing the options dataclass to not a dataclass, please fix that

**Comment by Exempt-Medic** [comment]
If you get this back to how the other PR was, the tags can be changed to just waiting on core since the previous one was

**Review by dontjoome — COMMENTED** [review]
should fix the issues

File: worlds/smz3/Options.py [hunk 1]
```diff
@@ -127,19 +128,21 @@ class EnergyBeep(DefaultOnToggle):
...
+    "key_shuffle": KeyShuffle,
+    "open_tower": OpenTower, 
+    "ganon_vulnerable": GanonVulnerable,
+    "open_tourian": OpenTourian,
+    "spin_jumps_animation": SpinJumpsAnimation,
+    "heart_beep_speed": HeartBeepSpeed,
+    "heart_color": HeartColor, 
+    "quick_swap": QuickSwap,
+    "energy_beep": EnergyBeep
+    }
```
Comment (Exempt-Medic) [inline worlds/smz3/Options.py:148]:
```suggestion
@dataclass
class SMZ3Options(PerGameCommonOptions):
    start_inventory_from_pool: StartInventoryPool
    accessibility: ItemsAccessibility
    sm_logic: SMLogic
    sword_location: SwordLocation
    morph_location: MorphLocation
    goal: Goal
    key_shuffle: KeyShuffle
    open_tower: OpenTower
    ganon_vulnerable: GanonVulnerable
    open_tourian: OpenTourian
    spin_jumps_animation: SpinJumpsAnimation
    heart_beep_speed: HeartBeepSpeed
    heart_color: HeartColor
    quick_swap: QuickSwap
    energy_beep: EnergyBeep
```

**Review by dontjoome — COMMENTED** [review]
:D derp, what I get for looking on my phone, so much tougher to see everything going on.

**Review by Exempt-Medic — APPROVED** [review]
Tested with `Missile: 40` and `Missile: 41` and the proper counts were grabbed and new items created

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**